File: 1832C.py
for _ in range(int(input())):
	N = int(input())
	A = [*map(int,input().split())]
	ans = unique(A)
	for i in range(N-2):
		if A[i]<A[i+1] and A[i+1]<A[i+2]:
			ans-=1
		if A[i]>A[i+1] and A[i+1]>A[i+2]:
			ans-=1	
	print(ans)

File: 1832C.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
  int t;
  cin >> t;
  while (t--) {
    int n;
    cin >> n;
    vector<int> a(n);
    for (int i = 0; i < n; ++i) {
      cin >> a[i];
    }

    n = unique(a.begin(), a.end()) - a.begin();
    int ans = n;
    for (int i = 0; i < n - 2; ++i) {
      if (a[i] < a[i + 1] && a[i + 1] < a[i + 2]) {
        --ans;
      }
      if (a[i] > a[i + 1] && a[i + 1] > a[i + 2]) {
        --ans;
      }
    }

    cout << ans << endl;
  }

  return 0;
}